templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/porter.yaml

```diff
@@ -221,3 +238,53 @@ stop:
         - stop
       flags:
         ids: "{{ bundle.outputs.azure_resource_id }}"
+
+reset_password:
+  - terraform:
+      arguments:
+        - "output"
+      description: "Get VM details from Terraform outputs"
+      backendConfig:
+        resource_group_name: "{{ bundle.parameters.tfstate_resource_group_name }}"
+        storage_account_name: "{{ bundle.parameters.tfstate_storage_account_name }}"
+        container_name: "{{ bundle.parameters.tfstate_container_name }}"
+        key: "{{ bundle.parameters.id }}"
+      outputs:
+        - name: azure_resource_id
+        - name: vm_username
+        - name: vm_password_secret_name
+        - name: keyvault_name
+  - az:
```

Comment:
@jjgriff93  If these steps were in a single shell script would that avoid the need to pass the secret between stages?

@tamirkamara what do you think?

Reply:
Working nicely now, no output so no risk of secrets leaking